autobahn/wamp/cryptosign.py

```diff
@@ -392,6 +399,36 @@ def format_challenge(session, challenge, channel_id_type) -> bytes:
 
         return data
 
+    def sign_challenge(data: bytes, signer_func: Callable):
```

Comment:
on a side note, I am wondering if there is an async callable type for the hinting .. as the callable must return a future (since the code doesn't use "maybe future" wrapper)

Reply:
There is no neat solution there, especially since our functions return `Deferred` OR `Future` depending on the txaio backend.

The general suggestion on the internet seems to be `Callable[[bytes], Awaitable[bytes]]`, which means that the parameter takes `bytes` as a first argument and returns an "awaitable" that resolves to `bytes`

Reply:
ok, I see, yes. without a proper advanced type system, it's hard to express the choice of network framework determines exactly which of Future or Deferred must be returned. A run-time cross-cutting concern. Anyways, it's cool for me! type hint and docs should be clear enough.

Reply:
> Callable[[bytes], Awaitable[bytes]]

yeah, I can see how that makes sense. however, it points to an even bigger elephant: `Callable`? great, `lambda: None` is certainly good then, right? ;) IOW: `Callable` is an incomplete type.